solvers/relay_constellation/umcf_srr_contact_plan/src/action_generation.py

```
@@ -0,0 +1,355 @@
+    umcf_instances
+        One per entry in ``sample_assignments``, used only for ``sample_index``.
+    sample_assignments
+        List of demand-to-path maps, one per UMCF instance.
+
+    Returns
+    -------
+    dict[tuple[str, str], set[int]]
+        Canonical edge -> sample indices where the edge appears in an SRR path.
+    """
+    edge_samples: dict[tuple[str, str], set[int]] = {}
+    for instance, assignments in zip(umcf_instances, sample_assignments):
```

Ruff (0.15.11)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)